cnd/src/bitcoin_fees.rs

```diff
@@ -11,3 +11,142 @@ impl BitcoinFees {
         Ok(self.0)
     }
 }
+
+mod blockcypher {
+    use ::bitcoin::Amount;
+    use anyhow::Result;
+    use serde::Deserialize;
+
+    #[derive(Debug, Clone)]
+    pub struct Client {
+        url: url::Url,
+        network: comit::Network,
+    }
+
+    /// https://www.blockcypher.com/dev/bitcoin/#blockchain
+    impl Client {
+        /// Full URL to access blockcypher API blockchain endpoint: http://api.blockcypher.com/v1/btc/main
+        pub fn new(url: url::Url, network: comit::Network) -> Self {
+            Self { url, network }
+        }
+
+        pub async fn get_fee_per_kb_as_per_expiries_recommendation(&self) -> Result<Amount> {
+            let recommendation = comit::expiries::bitcoin_mine_within_blocks(self.network);
+            if recommendation < 3 {
+                self.high_fee_per_kb().await
+            } else if recommendation < 7 {
+                self.medium_fee_per_kb().await
+            } else {
+                self.low_fee_per_kb().await
+            }
+        }
+
+        /// A rolling average of the fee (in satoshis) paid per kilobyte for
+        /// transactions to be confirmed within 1 to 2 blocks.
+        async fn high_fee_per_kb(&self) -> Result<Amount> {
+            let response = self.get().await?;
+
+            let fee = response.high_fee_per_kb;
+
+            tracing::info!(
```

Comment:
I appreciate your effort in writing more and good log statements!

In my experience though, having log statements within "connectors" or "clients" adds up to a lot of noise because they may end up being called more often than their data is actually used to make decisions.

Hence, a log statement that goes next to where we use this data would probably be more informative and also less noisy.

If you want keep it here, please reduce this to at least `debug` but probably actually `trace`.

For `info` level, my go to rule of thumb is that it should be associated with a "business state change", like: "Closed order XYZ in swap" or "Swap ABC completed successfully". Ideally, running the software on `info` level tells you about the progress it makes in doing what you want it to do. That only works if we are not too noisy on info level.

Note that I am describing an ideal scenario here, there are probably several log statements that _don't_ adhere to this at the moment :grimacing:

Reply:
In this particular instance, I think it can be kept and downgraded to `debug`.